### PR TITLE
MLT-41 Change handling of duplicate orders

### PR DESCRIPTION
**Pull request by MormonJesus69420**
This pull requests aims to make handling of duplicate orders similar to the way we handle duplicate products.
It comes with the same weakness in case SynQ informs us that the order is a duplicate, not our DB. Problem being we have no information about the order and no way of getting that information through API.

### CODE REVIEW

**Comment by MormonJesus69420** [comment]
I forgor about the API Doc...

File: src/main/kotlin/no/nb/mlt/wls/product/service/ProductService.kt [hunk 1]
```diff
@@ -47,7 +48,13 @@ class ProductService(val db: ProductRepository, val synqProductService: SynqProd
         val product = payload.toProduct().copy(quantity = 0.0, location = null)
 
         // Product service should create the product in the storage system, and return error message if it fails
-        if (synqProductService.createProduct(product.toSynqPayload()).statusCode.isSameCodeAs(HttpStatus.OK)) {
+        val synqResponse = synqProductService.createProduct(product.toSynqPayload())
+        // If SynQ didn't throw an error, but returned 4xx/5xx, then it is likely some error or edge-case we haven't handled
+        if (synqResponse.statusCode.is4xxClientError || synqResponse.statusCode.is5xxServerError) {
+            throw ServerErrorException("Unexpected error with SynQ", null)
+        }
```
Comment (anotheroneofthese) [inline src/main/kotlin/no/nb/mlt/wls/product/service/ProductService.kt:55]: This code might be redundant, as the WebClient should always map a 4xx or 5xx to an error, which is thrown

**Review by anotheroneofthese — COMMENTED** [review]
Looks good. Only a single comment

File: src/main/kotlin/no/nb/mlt/wls/product/service/ProductService.kt (hunk 1 above)
Comment (MormonJesus69420) [reply src/main/kotlin/no/nb/mlt/wls/product/service/ProductService.kt:55]: If that's so, we might use it to handle 3xx codes and/or codes other than 200 and 201?

Comment (anotheroneofthese) [reply src/main/kotlin/no/nb/mlt/wls/product/service/ProductService.kt:55]: I think codes other than what we expect (200 and 201) should be good

**Review by anotheroneofthese — APPROVED** [review]

**Pull request merged**